encoding/template/delta_iterator.go

```diff
@@ -0,0 +1,89 @@
+package template
+
+import (
+	bitstream "github.com/dgryski/go-bitstream"
+)
+
+type applyOpToValueIntFn func(v GenericValue, delta int) GenericValue
+
+// DeltaValueIterator iterates over a stream of delta encoded data.
+type DeltaValueIterator struct {
+	bitReader           *bitstream.BitReader
+	bitsPerEncodedValue int64
+	subFn               applyOpToValueIntFn
+	addFn               applyOpToValueIntFn
+	negativeBit         uint64
+	curr                GenericValue
+	err                 error
+	closed              bool
+	isFirstValue        bool
```

Comment:
nit: the convention is to make it so that you don't need to explicitly initialize it (i.e., the default value for bool should just work by changing `isFirstValue` to `isDeltaValue`)